### PR TITLE
Updating Schema

### PR DESCRIPTION
**Pull request by aqeelmozumder**
Have added changes for GenerateScheduleInput:

- Added Term
- Added Course (Not sure the type of this)

Have added changes for User:

- Added hasPeng (to match prisma)

Have added some changes required by front end:

**High Priority Changes:**

- Added EditCourse Mutation
- Added Section Number
- Added Non-teaching term
- Added Teaching capacity
- Added Has Relief
- Added Relief Explanation
- Added Has topic
- Added Topic Description
- Added Display Name

**Low Priority Changes:**

- Added Course Title
- Added Query to grab all course preferences
- Query for all users

### CODE REVIEW

File: src/schema/schema.graphql [hunk 1]
```diff
@@ -78,6 +86,10 @@ type Mutation {
   Generate schedule
   """
   generateSchedule(input: GenerateScheduleInput!): Response!
+  """
+  Edit schedule
+  """
+  EditSchedule(year: Int!): Response!
```
Comment (kendajani) [inline src/schema/schema.graphql:92]: Should we bother with this for this sprint?

File: src/schema/schema.graphql [hunk 2]
```diff
@@ -128,7 +146,9 @@ input CoursePreferenceInput {
 
 input GenerateScheduleInput {
   year: Int!
-  # TODO: need to fill in the rest of the inputs
+  term: Term!
+  courses: String
+  #not sure what type the course will be? CoursSection?
```
Comment (kendajani) [inline src/schema/schema.graphql:151]: Probably CourseSection? @aomi what do you think?

File: src/resolvers/index.ts [hunk 3]
```diff
@@ -61,6 +67,10 @@ export const resolvers: Resolvers<Context> = {
       if (!ctx.session.user) return null;
       return getSchedule(params.year || new Date().getFullYear());
     },
+    allUsers: async (_, _params, ctx) => {
+      if (!ctx.session.user || !ctx.session.user.username) return null;
```
Comment (aomi) [inline src/resolvers/index.ts:71]: don't we want to only allow this to be tied to admin?

**Review by aomi — COMMENTED** [review]
Are the failing tests fixable or unrelated to these changes?

**Comment by aqeelmozumder** [comment]
> Are the failing tests fixable or unrelated to these changes?

It is related to the schema changes but I would make a separate PR to fix the tests

**Review by aomi — APPROVED** [review]
LGTM

**Pull request merged**